web-console/src/utils/ingestion-spec.tsx

```diff
@@ -1173,11 +1174,34 @@ export function getIoConfigFormFields(ingestionComboType: IngestionComboType): F
           type: 'string-array',
           placeholder: 's3://your-bucket/some-path1, s3://your-bucket/some-path2',
           required: true,
-          defined: ioConfig => !deepGet(ioConfig, 'inputSource.uris'),
+          defined: ioConfig =>
+            !deepGet(ioConfig, 'inputSource.uris') && !deepGet(ioConfig, 'inputSource.objects'),
           info: (
             <>
               <p>A list of paths (with bucket) where your files are stored.</p>
-              <p>Either S3 URIs or S3 prefixes must be set.</p>
+              <p>Either S3 URIs or prefixes or objects must be set.</p>
+            </>
+          ),
+        },
+        {
+          name: 'inputSource.objects',
+          label: 'S3 objects',
+          type: 'json',
+          required: true,
```

Comment:
It would be cool to provide a `placeholder` to show what kind of shape of thing goes in here

Reply:
<img width="290" alt="Screen Shot 2019-12-05 at 1 40 28 PM" src="https://user-images.githubusercontent.com/1577461/70277358-17684080-1767-11ea-9eb4-9e6b41e0bee4.png">